### PR TITLE
US027: Apply get current cars action 

### PR DESCRIPTION
**Pull request by dilmurod-rozikov**

### CODE REVIEW

File: ParkingZoneApp.Tests/ModelValidation/ParkingSlots/FilterSlotVMTests.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    public class FilterSlotVMTests
+    {
+        public static IEnumerable<object[]> TestData =>
+           [
+                [Guid.NewGuid(), SlotCategory.Standard, true, true],
+           ];
+
+        [Theory]
+        [MemberData(nameof(TestData))]
+        public void GivenItemToBeValidated_WhenCreatingGetCurrentCarsVM_ThenValidationIsPerformed
```
Comment (SamDotNetDev) [inline ParkingZoneApp.Tests/ModelValidation/ParkingSlots/FilterSlotVMTests.cs]:
```suggestion
        public void GivenItemToBeValidated_WhenCreatingFilterSlotsVM_ThenValidationIsPerformed
```

File: ParkingZoneApp.Tests/ModelValidation/ParkingZones/GetCurrentCarsTests.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+using System.ComponentModel.DataAnnotations;
+
+namespace ParkingZoneApp.Tests.ModelValidation.ParkingZones
+{
+    public class GetCurrentCarsTests
+    {
+        public static IEnumerable<object[]> TestData =>
+           new List<object[]>
+           {
+                new object[] { 1, "Test1", true },
```
Comment (SamDotNetDev) [inline ParkingZoneApp.Tests/ModelValidation/ParkingZones/GetCurrentCarsTests.cs:11]: check for null also for VehicleNumber cuz string could be null

File: ParkingZoneApp/Services/ReservationService.cs [hunk 3]
```diff
@@ -13,5 +13,10 @@ public IEnumerable<Reservation> GetReservationsByUserId(string userId)
         {
             return GetAll().Where(x => x.UserId == userId);
         }
+
+        public IEnumerable<Reservation> GetReservationsByZoneId(Guid zoneId)
+        {
+            return GetAll().Where(x => x.ParkingZoneId == zoneId);
```
Comment (SamDotNetDev) [inline ParkingZoneApp/Services/ReservationService.cs]: logic is wrong you have to show current cars not all. Now you are showing even the cars which were in the parkingZone in the past. You have to take only cars which are currently in certain ParkingZone

**Review by SamDotNetDev — CHANGES_REQUESTED** [review]

File: ParkingZoneApp.Tests/ModelValidation/ParkingZones/GetCurrentCarsTests.cs (hunk 2 above)
Comment (dilmurod-rozikov) [reply ParkingZoneApp.Tests/ModelValidation/ParkingZones/GetCurrentCarsTests.cs:11]: I did and it is failing for no reason

Comment (SamDotNetDev) [reply ParkingZoneApp.Tests/ModelValidation/ParkingZones/GetCurrentCarsTests.cs:11]: try to solve it it has to check for null also

**Review by SamDotNetDev — APPROVED** [review]

**Review by abdullokhtolibjonov — APPROVED** [review]

**Pull request merged**